### PR TITLE
Fixes #15497: deactivate glob dumping while executing Load

### LINKED ISSUE #15497: .glob files are incorrect (and result in incorrect html linking) when `Load` is used
#### Description of the problem

Consider the following files:
`A.v`:
```coq
Require Import Init.
Require Import Init.
```
`B.v`:
```coq
Require Import Init. Local Open Scope nat_scope.
Require Import Init.
Load A.
Check nat.
```
`C.v`:
```coq
Require Import Init. Local Open Scope nat_scope.
Require Import Init.
(*****)
Check nat.
```
If I run `coq_makefile -Q . Top A.v B.v C.v -o Makefile` and then `make clean` and then `make all html`, I get the following `.glob` files:
`A.glob`:
```
DIGEST d59f3182ed63bdbcf24de6c7917f9699
FTop.A
R15:18 Coq.Classes.Init <> <> lib
R36:39 Coq.Classes.Init <> <> lib
```
`B.glob`:
```
DIGEST 4f20981d9fdb7f179fb549820cab8dea
FTop.B
R15:18 Coq.Classes.Init <> <> lib
R64:67 Coq.Classes.Init <> <> lib
R15:18 Coq.Classes.Init <> <> lib
R36:39 Coq.Classes.Init <> <> lib
R84:86 Coq.Init.Datatypes <> nat ind
```
`C.glob`:
```
DIGEST 5d05a02f5187592c07d44a384272f7f8
FTop.C
R15:18 Coq.Classes.Init <> <> lib
R64:67 Coq.Classes.Init <> <> lib
R84:86 Coq.Init.Datatypes <> nat ind
```

Note that `B.glob` should be the same as `C.glob`, but instead it has all of the references from `A.v`.  This means that the generated html contains incorrect links:
<details><summary>Top.B.html source</summary>

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link href="coqdoc.css" rel="stylesheet" type="text/css" />
<title>Top.B</title>
</head>

<body>

<div id="page">

<div id="header">
</div>

<div id="main">

<h1 class="libtitle">Library Top.B</h1>

<div class="code">
<span class="id" title="keyword">Require</span> <span class="id" title="keyword">Import</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">Init</span></a>. <span class="id" title="var">Local Open</span> <span class="id" title="keyword">Scope</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">nat_scope</span></a>.<br/>
<span class="id" title="keyword">Require</span> <span class="id" title="keyword">Import</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">Init</span></a>.<br/>
<span class="id" title="keyword">Load</span> <span class="id" title="var">A</span>.<br/>
<span class="id" title="keyword">Check</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Init.Datatypes.html#nat"><span class="id" title="inductive">nat</span></a>.<br/>
</div>
</div>

<div id="footer">
<hr/><a href="index.html">Index</a><hr/>This page has been generated by <a href="http://coq.inria.fr/">coqdoc</a>
</div>

</div>

</body>
</html>
```
</details>

<details><summary>Top.C.html source</summary>

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link href="coqdoc.css" rel="stylesheet" type="text/css" />
<title>Top.C</title>
</head>

<body>

<div id="page">

<div id="header">
</div>

<div id="main">

<h1 class="libtitle">Library Top.C</h1>

<div class="code">
<span class="id" title="keyword">Require</span> <span class="id" title="keyword">Import</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">Init</span></a>. <span class="id" title="var">Local Open</span> <span class="id" title="keyword">Scope</span> <span class="id" title="var">nat_scope</span>.<br/>
<span class="id" title="keyword">Require</span> <span class="id" title="keyword">Import</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">Init</span></a>.<br/>
<span class="id" title="keyword">Check</span> <a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Init.Datatypes.html#nat"><span class="id" title="inductive">nat</span></a>.<br/>
</div>
</div>

<div id="footer">
<hr/><a href="index.html">Index</a><hr/>This page has been generated by <a href="http://coq.inria.fr/">coqdoc</a>
</div>

</div>

</body>
</html>
```
</details>

`Top.B.html`:
image
vs
`Top.C.html`:
image
Note that `nat_scope` should not be linked (as it is not in C), but in B it is linked as `<a class="idref" href="http://coq.inria.fr/distrib/V8.14.0/stdlib//Coq.Classes.Init.html#"><span class="id" title="library">nat_scope</span></a>`.

#### Coq Version

8.14

### PR DESCRIPTION
**Pull request by herbelin**
**KInd:** fix

Fixes #15497 (part 1)

Note: a further PR implementing the wish part of #15497 will have a test.

- [ ] Added **changelog**.

### CODE REVIEW

File: vernac/vernacinterp.ml [hunk 1]
```diff
@@ -189,7 +189,9 @@ and vernac_load ~verbosely fname =
       let stack, pm = v_mod (interp_control ~st:{ st with Vernacstate.lemmas = stack; program = pm }) stm in
       (load_loop [@ocaml.tailcall]) ~stack ~pm
   in
+  let () = Dumpglob.(push_output NoGlob) in
   let stack, pm = load_loop ~pm:st.Vernacstate.program ~stack:st.Vernacstate.lemmas in
+  let () = Dumpglob.pop_output () in
```
Comment (SkySkimmer) [inline vernac/vernacinterp.ml:194]: If load raises the pop won't be executed.

**Review by SkySkimmer — CHANGES_REQUESTED** [review]

File: vernac/vernacinterp.ml (hunk 1 above)
Comment (herbelin) [reply vernac/vernacinterp.ml]: Right, thanks. Updated.

File: interp/dumpglob.ml [hunk 2]
```diff
@@ -41,6 +41,17 @@ let pop_output () = glob_output := match !glob_output with
 let pause () = push_output NoGlob
 let continue = pop_output
 
+let without_glob f () =
+  push_output NoGlob;
```
Comment (SkySkimmer) [inline interp/dumpglob.ml:45]:
```suggestion
let with_output g f () =
  push_output g;
```
?

Comment (herbelin) [reply interp/dumpglob.ml:45]: Yes, good idea. I'd just propose to name it `with_glob_output` since there is already a `Topfmt.with_output_to`??

Comment (SkySkimmer) [reply interp/dumpglob.ml:45]: Do as you prefer.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**